### PR TITLE
[Ballista] Enable ApproxPercentileWithWeight in Ballista and fill UT 

### LINKED ISSUE #2193: Enable ApproxPercentileWithWeight in Ballista
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
After fix #2180 , Enable ApproxPercentileWithWeight in Ballista.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #2193.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Ted-Jiang** [comment]
@houqp @yjshen plz take a look.

File: ballista/rust/client/src/context.rs [hunk 1]
```diff
@@ -718,6 +719,199 @@ mod tests {
...
+        assert_result_eq(expected, &*res);
+
+        let df = context
+            .sql("select approx_percentile_cont_with_weight(\"id\", 2, 0.5) from test")
+            .await
+            .unwrap();
+        let res = df.collect().await.unwrap();
+        let expected = vec![
+            "+---------------------------------------------------------------+",
+            "| APPROXPERCENTILECONTWITHWEIGHT(test.id,Int64(2),Float64(0.5)) |",
```
Comment (yjshen) [inline ballista/rust/client/src/context.rs:908]: 🤔 hard to read a little bit.

**Comment by liukun4515** [comment]
I think we should add the function in https://github.com/apache/arrow-datafusion/blob/28a6da3d2d175eb9d2f4ff8a6ea58e7c22dae97c/ballista/rust/core/proto/datafusion.proto#L203 proto

File: ballista/rust/client/src/context.rs (hunk 1 above)
Comment (Ted-Jiang) [reply ballista/rust/client/src/context.rs:908]: I thinks this name is align with proto file, it try to rewrite `debug` or `display` of this struct , it not works😂

**Comment by Ted-Jiang** [comment]
> I think we should add the function in
> 
> https://github.com/apache/arrow-datafusion/blob/28a6da3d2d175eb9d2f4ff8a6ea58e7c22dae97c/ballista/rust/core/proto/datafusion.proto#L203
> 
> proto

**Comment by Ted-Jiang** [comment]
> 

There is some issue in #2221

**Comment by Ted-Jiang** [comment]
> 

There is some issue in https://github.com/apache/arrow-datafusion/issues/2221

**Comment by Ted-Jiang** [comment]
> I think we should add the function in
> 
> https://github.com/apache/arrow-datafusion/blob/28a6da3d2d175eb9d2f4ff8a6ea58e7c22dae97c/ballista/rust/core/proto/datafusion.proto#L203
> 
> proto

Yes , but there is some issue in #2221

**Comment by liukun4515** [comment]
> > I think we should add the function in
> > https://github.com/apache/arrow-datafusion/blob/28a6da3d2d175eb9d2f4ff8a6ea58e7c22dae97c/ballista/rust/core/proto/datafusion.proto#L203
> > 
> > proto
> 
> Yes , but there is some issue in #2221 but in df it just rewrite `APPROX_MEDIAN` to `APPROX_PERCENTILE(0.5)`

My thought is that we should add 
```
APPROX_PERCENTILE_CONT_WITH_WEIGHT = 16;
```
behind the  `APPROX_MEDIAN=15; ` in the proto file.

**Review by liukun4515 — APPROVED** [review]

**Review by thinkharderdev — APPROVED** [review]

**Review by yjshen — APPROVED** [review]

**Pull request merged**